roles/lib_openshift/library/oc_project.py

```diff
@@ -1543,19 +1543,20 @@ def needs_update(self):
     def run_ansible(params, check_mode):
         '''run the idempotent ansible code'''
 
-        _ns = None
-        if params['node_selector'] is not None:
-            _ns = ','.join(params['node_selector'])
-
-        pconfig = ProjectConfig(params['name'],
-                                'None',
-                                params['kubeconfig'],
-                                {'admin': {'value': params['admin'], 'include': True},
-                                 'admin_role': {'value': params['admin_role'], 'include': True},
-                                 'description': {'value': params['description'], 'include': True},
-                                 'display_name': {'value': params['display_name'], 'include': True},
-                                 'node_selector': {'value': _ns, 'include': True},
-                                })
+        node_selector = ','.join(params.get('node_selector', [])) or None
```

Comment:
I'm not sure this code will work the same way.  I like the shorter notation but the `params` dictionary will always have a `'node_selector'`.  It will either be `None` or a `list` of `key=value` strings.  I believe this will fail when `None` is not iterable.

`>>> params = {'node_selector': None}`
`>>> node_selector = ','.join(params.get('node_selector', [])) or None`
`Traceback (most recent call last):`
 ` File "<stdin>", line 1, in <module>`
`TypeError: can only join an iterable`